### PR TITLE
131 reduce api calls

### PR DESCRIPTION
**Pull request by jimmyko360**
Refactored function that makes my GET requests to use async/await
- front end carousels function similarly to before but using much fewer lines of code
- removed many extra lines of code that were redundant or unused, possibly causing my code to slow down

### CODE REVIEW

File: client/components/relatedWidget/Outfit.jsx [hunk 1]
```diff
@@ -19,17 +19,9 @@ class Outfit extends React.Component {
...
-      selected: selected
+      image: []
     };
-    this.productItems = ProductList(
-      {
-        products: this.props.yourOutfit,
-        images: this.state.image
-      },
-        this.state.selected,
-        this.props.getNewProduct);
```
Comment (jimmyko360) [inline client/components/relatedWidget/Outfit.jsx:31]: Keeping this process contained to componentDidUpdate

File: client/components/relatedWidget/Outfit.jsx [hunk 2]
```diff
@@ -44,30 +36,20 @@ class Outfit extends React.Component {
     });
   }
 
-  onSelect(key) {
-    this.setState({ selected: key });
-  }
-
```
Comment (jimmyko360) [inline client/components/relatedWidget/Outfit.jsx:50]: Function used to highlight product cards that isn't necessarily needed right now

File: client/components/relatedWidget/Outfit.jsx [hunk 3]
```diff
@@ -44,30 +36,20 @@ class Outfit extends React.Component {
...
     },
-      this.state.selected,
       this.props.getNewProduct);
   }
 
-  componentDidMount() {
-    this.getImage(this.props.product.id);
-    this.buildCarousel();
-  }
-
```
Comment (jimmyko360) [inline client/components/relatedWidget/Outfit.jsx:65]: Removed componentDidMount since it usually fires before it receives product.id which is undesired

**Review by rhunce — APPROVED** [review]

**Review by cchauche — APPROVED** [review]
As far as I can tell it does seem to do what is promised, it all seems to play nice on my side.

**Pull request merged**